tortoise/state.go

```diff
@@ -69,6 +70,10 @@ type (
 
 		// to efficiently find base and reference ballots
 		ballotRefs map[types.BallotID]*ballotInfo
+
+		// malnodes is a collection with all nodes that equivocated in history.
+		// each node id is 32 bytes. 100 000 of such nodes is only about ~3MB
+		malnodes map[types.NodeID]struct{}
```

Comment:
this seems parallel with the lru cache in CachedDB, and we are spending 2X memory for the malicious IDs. is there any way to combine?

Reply:
i understand that, but i don't see how to dedup.
i don't want to add dependency to that cache, and using tortoise to query for malfeasant ids will conflict with executing important things. would rather think that about memory if number of that ids will grow significantly

Reply:
i can actually add another rwlock for identities, and then every module can query that